### ISSUE TITLE
Align validations and API responses

### ISSUE THREAD

**Issue by kiramclean** [Swirrl/datahost-prototypes#189]
The API currently returns misleading responses if resources are created with missing required fields.
TL;DR proposed solution: Audit validations and `get-<resource>` queries to align required and optional fields. I propose making fields required rather than loosening the queries.

**More details:**
- Make title and description required on create for dataset series. This will make the query work as expected.
- Make release title and description required on create
- Make revision title required on create
- Make change (append) description required on create and update swagger UI to include a field to supply it

**Even more details:**
To reproduce the bugs:

Create a dataset with no title or description
```
curl -X 'PUT' \
  'http://localhost:3000/data/new-dataset' \
  -H 'accept: application/json'
```
Returns 201 (correctly):
```
{
  "@id" : "new-dataset",
  "@type" : "dh:DatasetSeries",
  "dcterms:issued" : "2023-08-02T02:01:08.982414Z",
  "dcterms:modified" : "2023-08-02T02:01:08.982414Z",
  "dh:baseEntity" : "https://example.org/data/new-dataset",
  "@context" : {
    "dh" : "https://publishmydata.com/def/datahost/",
    "dcterms" : "http://purl.org/dc/terms/",
    "rdf" : "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
    "@base" : "https://example.org/data/"
  }
```
but fetching it fails:
```
curl -X 'GET' \
  'http://localhost:3000/data/new-dataset' \
  -H 'accept: application/json'
```
because the title is required in the query but not on create. Same thing can be observed for other resources when leaving the title/description fields blank.

**Comment by rosado** [comment]
For series, releases and revisions we allow the user to supply: JSON body and/or query params. Both of those have different schemas at the moment (query params being optional). When supplied, query params override the values in the JSON body. 

One of the use cases is update of an existing resource, where supplying just the title makes sense. But that gets in the way of validation (we can deal with it with a custom middleware, though).

My question is: do we want to allow creation of resources using only the query params?

**Comment by rosado** [comment]
I used the [hurl](https://hurl.dev) file below to verify everything is working.

```
PUT http://localhost:3000/data/failed-dataset-013

HTTP 406

PUT http://localhost:3000/data/failed-dataset-013?description=failed

HTTP 406

PUT http://localhost:3000/data/failed-dataset-013?description=failed
Content-Type: application/json

HTTP 400

PUT http://localhost:3000/data/failed-dataset-013?description=failed&title=new-dataset
Content-Type: application/json

HTTP 201

PUT http://localhost:3000/data/failed-dataset-013-x
Content-Type: application/json
{
	"dcterms:title": "Series X",
	"dcterms:description": "A little test series"
}

HTTP 201

```
